src/proteus/dswx_hls.py

```diff
@@ -648,63 +750,69 @@ def create_landcover_mask(copernicus_landcover_file,
     # Reproject Copernicus land cover
     copernicus_landcover_reprojected_file = os.path.join(
         scratch_dir, 'copernicus_reprojected.tif')
-    copernicus_landcover_array = _relocate(copernicus_landcover_file,
+    copernicus_landcover_array = _warp(copernicus_landcover_file,
         geotransform, projection,
         length, width, scratch_dir, resample_algorithm='nearest',
         relocated_file=copernicus_landcover_reprojected_file,
         temp_files_list=temp_files_list)
     temp_files_list.append(copernicus_landcover_reprojected_file)
 
-    # Reproject ESA Worldcover 10m
+    # Reproject ESA Worldcover 10m from geographic (lat/lon) to MGRS (UTM) 10m
     geotransform_up_3 = list(geotransform)
     geotransform_up_3[1] = geotransform[1] / 3  # dx / 3
     geotransform_up_3[5] = geotransform[5] / 3  # dy / 3
     length_up_3 = 3 * length
     width_up_3 = 3 * width
     worldcover_reprojected_up_3_file = os.path.join(
         scratch_dir, 'worldcover_reprojected_up_3.tif')
-    worldcover_array_up_3 = _relocate(worldcover_file, geotransform_up_3,
+    worldcover_array_up_3 = _warp(worldcover_file, geotransform_up_3,
         projection, length_up_3, width_up_3, scratch_dir,
         resample_algorithm='nearest',
         relocated_file=worldcover_reprojected_up_3_file,
         temp_files_list=temp_files_list)
-    temp_files_list.append(worldcover_array_up_3)
+    temp_files_list.append(worldcover_reprojected_up_3_file)
 
     # Set multilooking parameters
     size_y = 3
     size_x = 3
 
     # Create water mask
-    logger.info(f'Creating water mask')
+    logger.info(f'creating water mask')
+    # WorldCover class 80: permanent water bodies
+    # WorldCover class 90: herbaceous wetland
+    # WorldCover class 95: mangroves
     water_binary_mask = np.where((worldcover_array_up_3 == 80) |
-                                 (worldcover_array_up_3 == 90), 1, 0)
-    h20_aggregate_sum = decimate_by_summation(water_binary_mask,
+                                 (worldcover_array_up_3 == 90) |
+                                 (worldcover_array_up_3 == 95), 1, 0)
```

Comment:
I'm not sure if this is faster (or more efficient). Notice that if the condition matches one element, the `np.where()` will return a single index. The code that you are suggesting will return an array of `bool`.

Reply:
Before making the suggestion, I did a quick test with the existing code, and it is slightly faster. (20 iterations of this line took ~11 sec for the np.where() version, vs ~9 sec with the .astype() version.) Not a huge deal for an isolated instance like this, but since small things add up over the course of the codebase, still good to keep in mind.

Also, by using .astype(int), it should return the array of 1s and 0s as desired:

```
>>> x = np.array([[1,2,3],[4,3,2]])
>>> z = (x>2).astype(int)
>>> z
array([[0, 0, 1],
       [1, 1, 0]])
```

Reply:
Thank you for testing, @nemo794 . I'm still not convinced that `astype(int)` is  better. Benchmarks can be a bit tricky because of caching. Another thing that I forgot to mention is the memory usage. The condition should match a low percentage of the input pixels (as we are looking for certain landcover classes). So, the size of the element returned by `np.where()` should be much smaller than the one returned by `astype(int)` , which BTW probably uses much more memory than `bool` (that I initially had in mind). The resulting array is passed to another function, so I'm not sure if using `astype(int)` is really more efficient?

Reply:
I just tested again, and np.where() is actually returning an int64 array as well:

Debug Code:
```
    import time
    start = time.time()
    for i in range(20):
        water_binary_mask = np.where((worldcover_array_up_3 == 80) |
                                    (worldcover_array_up_3 == 90) |
                                    (worldcover_array_up_3 == 95), 1, 0)
    print('water_binary_mask time (sec): ', time.time() - start)
    print('water_binary_mask.dtype: ', water_binary_mask.dtype)
    print("Memory size of one array element of water_binary_mask in bytes: ", water_binary_mask.itemsize)
    print("Memory size of numpy array water_binary_mask in bytes:", water_binary_mask.size * water_binary_mask.itemsize)
 
    start = time.time()
    for i in range(20):
        water_binary_mask2 = ((worldcover_array_up_3 == 80) |
                                    (worldcover_array_up_3 == 90) |
                                    (worldcover_array_up_3 == 95)).astype(int)
    print('water_binary_mask2 time (sec): ', time.time() - start)
    print('water_binary_mask2.dtype: ', water_binary_mask2.dtype)
    print("Memory size of one array element of water_binary_mask2 in bytes: ", water_binary_mask2.itemsize)
    print("Memory size of numpy array water_binary_mask2 in bytes:", water_binary_mask2.size * water_binary_mask2.itemsize)
```

Output:
```
water_binary_mask time (sec):  21.7958083152771
water_binary_mask.dtype:  int64
Memory size of one array element of water_binary_mask in bytes:  8
Memory size of numpy array water_binary_mask in bytes: 964483200

water_binary_mask2 time (sec):  16.826853275299072
water_binary_mask2.dtype:  int64
Memory size of one array element of water_binary_mask2 in bytes:  8
Memory size of numpy array water_binary_mask2 in bytes: 964483200
```

Is there a different/better way to test this? Also, I'm curious why passing it to another array would make a difference in efficiency? (Does astype() do something funny under the hood?)

Reply:
Uhm, that's right, the code is not doing what I expected. I just did a quick test here and I noticed the following:

```
>>> np.where([1, 0, 0])
(array([0]),)
>>> np.where([1, 0, 0], 1, 0)
array([1, 0, 0])
```

Notice that the first command returns only the indices that match the condition and the second returns the full array. This happens because of the extra arguments `, 1, 0)`. I copied that part of the code from John's script and I didn't pay attention to it. I also noticed now that the code inside the called function requires an array, rather than list of indices. So yes, it makes sense to change the masking. I'll update the code soon. Thank you.

Reply:
Thank you again for finding this opportunity to improve the code. I just substituted `astype(int)`  with `astype(np.uint8)`. The commits that contain the changes are the following: [2bc5137](https://github.com/nasa/PROTEUS/pull/28/commits/2bc5137ac7c372c76a6da4bd48e409c5abd92481) and [5be73c2](https://github.com/nasa/PROTEUS/pull/28/commits/5be73c21ab2c47d001c9cd51d2545586a4264635) .

Reply:
Ah, that explains one of the comments from yesterday. (tbh, seeing the `, 1, 0)` was what caught my eye in the first place, because it felt like there could be a more Pythonic way.)

Glad we got to the bottom of it, and thank you for making these updates and the shift to `np.uint8`.

Looks good!